operator/pkg/upgrade/upgrade.go

```diff
@@ -31,6 +32,11 @@ func Upgrade(ctx context.Context, cli client.Client, in *clusterv1beta1.Installa
 		return fmt.Errorf("k0s upgrade: %w", err)
 	}
 
+	err = registryMigrationStatus(ctx, cli, in)
```

Comment:
why would there be a registry migration during an upgrade?

Reply:
new installation objects don't get the registry migration status copied over

Reply:
and they shouldn't, no?

Reply:
it needs it else there is a bug explained here https://github.com/replicatedhq/embedded-cluster/pull/1238#discussion_r1793917391

Reply:
i see. seems like a bandaid covering a bigger issue in the reconciliation process.

Reply:
i wonder if it should be preserved by whatever creates the new installation object though like everything else

Reply:
ya i agree its a bandaid. i did create this. i think we could improve the logic here

https://app.shortcut.com/replicated/story/113699/it-is-not-possible-to-migrate-to-high-availability-after-a-restore